code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -60,10 +60,16 @@
 		item_state = "flamethrower_0"
 	return
 
-/obj/item/weapon/flamethrower/afterattack(atom/target, mob/user, flag)
-	if(flag) return // too close
-	// Make sure our user is still holding us
-	if(user && user.get_active_held_item() == src)
+/obj/item/weapon/flamethrower/afterattack(atom/target, mob/living/carbon/human/user, flag)
```

Comment:
this doesn't actually enforce anything kor, so it will actually cause runtimes when non carbons are passed through since they don't always have dna or traits.